lms/static/js/student_account/views/account_settings_fields.js

```diff
@@ -1,191 +1,233 @@
+                        attributes[this.options.valueAttribute] = value;
+                        this.saveAttributes(attributes);
+                    }
                 }
-            },
-
-            redirect_to: function (url) {
-                window.location.href = url;
-            },
-
-            disconnect: function () {
-                var data = {};
-
-                // Disconnects the provider from the user's edX account.
-                // See python-social-auth docs for more information.
-                var view = this;
-                $.ajax({
-                    type: 'POST',
-                    url: this.options.disconnectUrl,
-                    data: data,
-                    dataType: 'html',
-                    success: function () {
-                        view.options.connected = false;
-                        view.render();
-                        view.showSuccessMessage();
-                    },
-                    error: function (xhr) {
-                        view.showErrorMessage(xhr);
+            }),
+            AuthFieldView: FieldViews.LinkFieldView.extend({
+                fieldTemplate: field_social_link_template,
+                className: function () {
+                    return 'u-field u-field-social u-field-' + this.options.valueAttribute;
+                },
+                initialize: function (options) {
+                    this.options = _.extend({}, options);
+                    this._super(options);
+                    _.bindAll(this, 'redirect_to', 'disconnect', 'successMessage', 'inProgressMessage');
+                },
+                render: function () {
+                    var linkTitle = '',
+                        linkClass = '',
+                        subTitle = '',
+                        screenReaderTitle = window.interpolate_text(
+                            gettext("Link your {accountName} account"), {accountName: this.options.title}
+                        );
+                    if (this.options.connected) {
+                        linkTitle = gettext('Unlink This Account');
+                        linkClass = 'social-field-linked';
+                        subTitle = window.interpolate_text(
+                            gettext('You can use your {accountName} account to sign in to your edX account.'),
+                            {accountName: this.options.title}
+                        );
+                        screenReaderTitle = window.interpolate_text(
+                            gettext("Unlink your {accountName} account"), {accountName: this.options.title}
+                        );
+                    } else if (this.options.acceptsLogins) {
+                        linkTitle = gettext('Link Your Account');
+                        linkClass = 'social-field-unlinked';
+                        subTitle = window.interpolate_text(
+                            gettext(
```

Comment:
@waheedahmed This message should not be split across multiple lines and must not include "edX".